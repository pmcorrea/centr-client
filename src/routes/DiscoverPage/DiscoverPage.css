.discover_ul {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100vw;
  padding-bottom: 80px;
  padding-left: 0px;
  margin: 0 auto;
}

.discover_li {
  display: flex;
  text-decoration: none;
  color: black;
  width: 100%;
}

.connection_name {
  font-size: 20px;
  color: white;
  margin: 0px;
  margin-top: 10px;
}

.approve_button {
  position: relative;
  border: none;
  border-radius: 0px;
  background-color: black;
  color: white;
  padding: 0px;
}

.discover_follow_button {
  position: relative;
  border: 2px solid rgb(112, 39, 161);
  border-radius: 50%;
  background-color: rgb(112, 39, 161);
  color: white;
  margin: 0px;
  height: 30px;
  width: 30px;
  font-size: 15px;
  font-weight: bold;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 0px;
  box-shadow: none;
}

.discover_follow_button:hover {
  background-color: orange;
  border: orange;
  color: white;
}

.user_visibility {
  color: black;
}

.avatar {
  width: 60px;
  height: 60px;
  border-radius: 50%;
}
