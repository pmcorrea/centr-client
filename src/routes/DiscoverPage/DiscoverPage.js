import React, { Component } from "react";
import "./DiscoverPage.css";
import AuthApiService from "../../services/auth-api-service";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";


export default class DiscoverPage extends Component {

	state = {
		users: null,
		error: null,
		currentUser: null,
		notFollowing:null
		
	}

	componentDidMount() {
		this.setCurrentUser()
		this.setNotFollowing()
	}

	setNotFollowing() {
		AuthApiService.getNotFollowing()
		.then(notFollowing => {
			if (notFollowing) {
				this.setState({
					notFollowing: notFollowing
				})
			}

		})
		.catch(error => {
			this.setState({ error }, () => console.log(this.state.error))
		})
	}

	setCurrentUser() {
		AuthApiService.getCurrentUserDetailsById()
		.then(currentUser => {
			currentUser = currentUser[0]

			this.setState({
				currentUser: currentUser
			})
		})
		.catch(error => {
			this.setState({ error }, () => console.log(this.state.error))
		})
	}

	sendFollowRequest(toUser) {
		let userId = toUser.id

		AuthApiService.postRequest(userId)
		.then(result => {
			this.setNotFollowing()
		})
		.catch(error => {
			this.setState({ error }, () => console.log(this.state.error))
		})
	}

  render() {

	let notFollowing = this.state.notFollowing ? this.state.notFollowing : ('')
      
    return this.state.notFollowing ? (
	<>
		<ul className="discover_ul">
		<p className="section_headers">Discover</p> 
			{notFollowing.map(user => (
				<li className="discover_li" key={user.user_name}>	
					<div className="discover_box">
						<img src={user.avatar} className="avatar" alt="avatar"></img>
						
						<h2 className="connection_name">
							{user.user_name}
						</h2>
					
						<button className='discover_follow_button' 
							onClick={() => {
								this.sendFollowRequest(user)
							}}>
							<FontAwesomeIcon icon='plus' size="1x"/>
						</button>
					</div>
				</li>
			))}
		</ul>
	</>	
    ) : ('')
	}
}