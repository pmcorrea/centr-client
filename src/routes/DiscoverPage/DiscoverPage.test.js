import DiscoverPage from './DiscoverPage.js'
import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App test suite', () => {
	// Smoke Test
	it('renders without crashing', () => {
		const div = document.createElement('div')

		ReactDOM.render(
			<DiscoverPage />,
			div
		)
		ReactDOM.unmountComponentAtNode(div)
	})

	// Snapshot Test
	it('renders the snapshot', () => {
		const wrapper = shallow(<DiscoverPage />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})