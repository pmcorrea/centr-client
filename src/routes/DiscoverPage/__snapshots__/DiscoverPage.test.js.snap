// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders the snapshot 1`] = `""`;
